packages/api/__tests__/specs/navigation.ts

```diff
@@ -0,0 +1,223 @@
+    })
+
+    test('can be created', async () => {
+      const {mutate} = testClientPrivate
+      const input: NavigationInput = {
+        name: 'Create Navigation Name',
+        key: 'createNavKey321',
+        links: [
+          {external: {label: 'External Label', url: 'linkurl.ch/'}},
+          {article: {label: 'Article Label', articleID: articleID}},
+          {page: {label: 'Page Label', pageID: pageID}}
+        ]
+      }
+      const res = await mutate({
+        mutation: CreateNavigation,
+        variables: {
+          input: input
+        }
+      })
+      expect(res).toMatchSnapshot({
+        data: {
+          createNavigation: {
+            id: expect.any(String),
+            links: expect.any(Array)
+          }
+        }
+      })
+
+      const links = res.data?.createNavigation?.links
+      expect(links).toHaveLength(3)
+      //external link
+      expect(links[0].label).toBe('External Label')
+      expect(links[0].url).toBe('linkurl.ch/')
+      //article
+      expect(links[1].label).toBe('Article Label')
+      expect(links[1].article.id).toBe(articleID)
+      //page
+      expect(links[2].label).toBe('Page Label')
+      expect(links[2].page.id).toBe(pageID)
+
+      ids.unshift(res.data?.createNavigation?.id)
+    })
+
+    test('can be read in list', async () => {
+      const {query} = testClientPrivate
+      const res = await query({
+        query: NavigationList
+      })
+      expect(res).toMatchSnapshot({
+        data: {
+          navigations: Array.from({length: ids.length}, () => ({
+            id: expect.any(String),
+            links: expect.any(Array)
+          }))
+        }
+      })
+      expect(res.data?.navigations?.length).toBe(ids.length)
+    })
+
+    test('can be read by id', async () => {
+      const {query} = testClientPrivate
+      const res = await query({
+        query: Navigation,
+        variables: {
+          id: ids[0]
+        }
+      })
+      expect(res).toMatchSnapshot({
+        data: {
+          navigation: {
+            id: expect.any(String)
+          }
+        }
+      })
+      expect(res.data?.navigation.id).toBe(ids[0])
+    })
+
+    test('can be updated', async () => {
+      const {mutate} = testClientPrivate
+      const res = await mutate({
+        mutation: UpdateNavigation,
+        variables: {
+          input: {
+            name: 'Updated Navigation Name',
+            key: 'updatedNavKey321',
+            links: [
+              {external: {label: 'New External Label', url: 'newlinkurl.ch/'}},
+              {article: {label: 'New Article Label', articleID: 'newID'}},
+              {page: {label: 'New Page Label', pageID: 'newID'}}
+            ]
+          },
+          id: ids[0]
+        }
+      })
+      expect(res).toMatchSnapshot({
+        data: {
+          updateNavigation: {
+            id: expect.any(String)
+          }
+        }
+      })
+      expect(res.data?.updateNavigation.id).toBe(ids[0])
```

Comment:
I would test here if the updated data is updated. for example check if the label says `New Page Label`